test/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java

```diff
@@ -342,6 +346,41 @@ public void userCreationWithHashedPasswords() throws Exception {
         assertTrue(spySecurityListener.createdUsers.get(0).equals("charlie_hashed"));
     }
 
+    @Issue("JENKINS-56008")
+    @Test
+    public void updateUserPassword() throws Exception {
+        HudsonPrivateSecurityRealm securityRealm = new HudsonPrivateSecurityRealm(true, false, null);
+        j.jenkins.setSecurityRealm(securityRealm);
+        WebClient wc = j.createWebClient();
+
+        spySecurityListener.usersWithPasswordUpdate.clear();
+        assertTrue(spySecurityListener.usersWithPasswordUpdate.isEmpty());
+
+        // new user account creation
+        SignupPage signup = new SignupPage(wc.goTo("signup"));
+        signup.enterUsername("debbie");
+        signup.enterPassword("debbie");
+        signup.enterFullName(StringUtils.capitalize("debbie user"));
+        signup.enterEmail("debbie" + "@" + "debbie" + ".com");
+        HtmlPage p = signup.submit(j);
+
+        assertEquals(200, p.getWebResponse().getStatusCode());
+
+        // execute an http request to change a user's password from their config page
+        User debbie = User.getById("debbie", false);
+        URL configPage = wc.createCrumbedUrl(debbie.getUrl() + "/" + "configSubmit");
+        String formData = "{\"fullName\": \"debbie user\", \"description\": \"\", \"userProperty3\": {\"primaryViewName\": \"\"}, \"userProperty5\": {\"password\": \"admin\", \"$redact\": [\"password\", \"password2\"], \"password2\": \"admin\"}, \"userProperty6\": {\"authorizedKeys\": \"\"}, \"userProperty8\": {\"insensitiveSearch\": true}, \"core:apply\": \"true\"}";
+
+        WebRequest request = new WebRequest(configPage, HttpMethod.POST);
+        request.setAdditionalHeader("Content-Type", "application/x-www-form-urlencoded");
+        request.setRequestBody("json=" + URLEncoder.encode(formData, StandardCharsets.UTF_8.name()));
```

Comment:
I can't believe this is how form encoding actually works. Ugh. :/

Reply:
lol